packages/keystatic/src/form/fields/markdoc/editor/Toolbar.tsx

```diff
@@ -534,61 +515,94 @@ function ListButtons() {
+
+  const disabledKeys = useMemo(() => {
+    return [
+      !canWrapInOrderedList && 'ordered_list',
+      !canWrapInUnorderedList && 'unordered_list',
+    ].filter(removeFalse);
+  }, [canWrapInOrderedList, canWrapInUnorderedList]);
+
   return useMemo(() => {
+    if (items.length === 0) {
+      return null;
+    }
+
     return (
-      <ActionGroup
-        flexShrink={0}
+      <EditorToolbarGroup
         aria-label="Lists"
-        selectionMode="single"
-        buttonLabelBehavior="hide"
-        density="compact"
-        prominence="low"
-        disabledKeys={[
-          !canWrapInOrderedList && 'ordered',
-          !canWrapInUnorderedList && 'unordered',
-        ].filter(removeFalse)}
-        summaryIcon={<Icon src={listIcon} />}
-        selectedKeys={activeListType ? [activeListType] : []}
-        onAction={key => {
-          const format = key as 'ordered' | 'unordered';
-          const type = schema.nodes[`${format}_list`];
+        value={activeListType}
+        onChange={key => {
+          const format = key as 'ordered_list' | 'unordered_list';
+          const type = schema.nodes[format];
           if (type) {
             dispatchCommand(toggleList(type));
           }
         }}
-        items={[
-          !!schema.nodes.unordered_list && {
-            label: 'Bullet List',
-            key: 'unordered',
-            shortcut: '-',
-            icon: listIcon,
-          },
-          !!schema.nodes.unordered_list && {
-            label: 'Numbered List',
-            key: 'ordered',
-            shortcut: '1.',
-            icon: listOrderedIcon,
-          },
-        ].filter(removeFalse)}
+        disabledKeys={disabledKeys}
+        selectionMode="single"
       >
-        {item => (
-          <Item textValue={`${item.label} (${item.shortcut})`}>
-            <Icon src={item.icon} />
-            <Text>{item.label}</Text>
-            <Kbd>{item.shortcut}</Kbd>
-          </Item>
-        )}
-      </ActionGroup>
+        {items.map(item => (
+          <TooltipTrigger key={item.key}>
+            <EditorToolbarItem value={item.key} aria-label={item.label}>
+              <Icon src={item.icon} />
+            </EditorToolbarItem>
+            <Tooltip>
+              <Text>{item.label}</Text>
+              <Kbd meta>{item.shortcut}</Kbd>
+            </Tooltip>
+          </TooltipTrigger>
+        ))}
+      </EditorToolbarGroup>
     );
-  }, [
-    activeListType,
-    canWrapInOrderedList,
-    canWrapInUnorderedList,
-    dispatchCommand,
-    schema.nodes,
-  ]);
+  }, [activeListType, disabledKeys, dispatchCommand, items, schema.nodes]);
 }
 
 function removeFalse<T>(val: T): val is Exclude<T, false> {
   return val !== false;
 }
+
+function removeAllMarks(): Command {
+  return (state, dispatch) => {
+    if (state.selection.empty) {
+      return false;
+    }
+
+    if (dispatch) {
+      // TODO: extend the range to include the whole mark on either side of the selection
```

Comment:
```suggestion
```

As discussed, the current behaviour is right, we don't want the behaviour described in the TODO